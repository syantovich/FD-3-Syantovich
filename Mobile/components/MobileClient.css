div.MobileClient
{
  padding: 2px 0;
}

span.MobileClientBalance
{
  float: right;
  width: 30px;
  font-weight: bold;
  text-align: right;
}

td.MobileClientFIO,th
{
  width: 150px;
  margin: 30px;
  padding: 0px;
  text-align: left;
}
td.active{
  background-color: green;
}
td.blocked{
  background-color: red;
}