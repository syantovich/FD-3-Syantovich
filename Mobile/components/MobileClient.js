import React from 'react';
import PropTypes from 'prop-types';
import {voteEvents} from "./events";



class MobileClient extends React.PureComponent {

  static propTypes = {
    client:PropTypes.shape({
      id: PropTypes.number.isRequired,
      FIO:PropTypes.shape({
        fam: PropTypes.string.isRequired,
        im: PropTypes.string.isRequired,
        otch: PropTypes.string.isRequired,
      }),
      balance: PropTypes.number.isRequired,
    })

  };

  state = {
    status:(this.props.client.balance<0)?"blocked":"active",
  };
  UNSAFE_componentWillReceiveProps=(nextProps)=> {
this.setState({status:(nextProps.client.balance<0)?"blocked":"active",})
}

  render() {

    console.log("MobileClient id="+this.props.client.id+" render");
    
    return (
        <tbody>
      <tr className='MobileClient'>

        <td className='MobileClientFIO'>{this.props.client.im}</td>
        <td className='MobileClientFIO'>{this.props.client.fam}</td>
        <td className='MobileClientFIO'>{this.props.client.otch}</td>
        <td className='MobileClientFIO'>{this.props.client.balance}</td>
        <td className={'MobileClientFIO '+this.state.status}>{this.state.status}</td>
        <td className='MobileClientFIO'><input type="button" value="Edit" onClick={()=>{voteEvents.emit("EsetMode","Edit",this.props.client.id)}}/></td>
        <td className='MobileClientFIO'><input type="button" value="Delete" onClick={()=>{voteEvents.emit("Edelete",this.props.client.id)}}/></td>
      </tr>
        </tbody>
    );

  }

}

export default MobileClient;
