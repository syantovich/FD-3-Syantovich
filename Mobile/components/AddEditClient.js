import React from 'react';
import PropTypes from 'prop-types';
import {voteEvents} from "./events";

import "./MobileClient.css"

class AddEditClient extends React.PureComponent {

    static propTypes = {
        clients:PropTypes.shape({
                id: PropTypes.number.isRequired,
                fam: PropTypes.string.isRequired,
                im: PropTypes.string.isRequired,
                otch: PropTypes.string.isRequired,
                balance: PropTypes.number.isRequired,
            }),
        workmode: PropTypes.string.isRequired,
        edited: PropTypes.number,
        maxId: PropTypes.number.isRequired,

    };
    constructor(props) {
        super(props);
        this.inId=React.createRef();
        this.inName=React.createRef();
        this.inFam=React.createRef();
        this.inOtch=React.createRef();
        this.inBalance=React.createRef();


    }

    state = {
        status:(this.props.client.balance<0)?"blocked":"active",
    };
    UNSAFE_componentWillReceiveProps=(nextProps)=> {
        this.setState({status:(nextProps.client.balance<0)?"blocked":"active",})
    }

    render() {
let id=this.props.client.id||this.props.maxId
        console.log("AddEdited id="+id+" render");
let ret;
if(this.props.workmode=="Show"){
    ret=<input type="button" value="Добавить клиента" id={"AddItem"} onClick={()=>{voteEvents.emit("EsetMode","Add",null)}}/>
}else{
    console.log(this.inId.current);
    console.log(this.inId);
    ret=<div key={this.props.edited||this.props.maxId}>
        id:<span ref={this.inId}>{this.props.edited||this.props.maxId}</span><br/>
        <label>Имя:<input type="text" defaultValue={(!!this.props.client)?this.props.client.im:""} ref={this.inName} id={'im'}/></label><br/>
        <label>Фамилия:<input type="text" defaultValue={(!!this.props.client)?this.props.client.fam:""} ref={this.inFam} id={"fam"}/></label><br/>
        <label>Отчество:<input type="text" defaultValue={(!!this.props.client)?this.props.client.otch:""} ref={this.inOtch} id={"otch"}/></label><br/>
        <label>Баланс:<input type="number" defaultValue={(!!this.props.client)?this.props.client.balance:0} ref={this.inBalance} id={"balance"}/></label><br/>
        {(this.props.workmode=="Edit")&&<input type="button" value="Save" id={"SaveEdit"} onClick={()=>voteEvents.emit("Esave",parseInt(this.inId.current.value),this.inName.current.value,this.inFam.current.value,this.inOtch.current.value,parseInt(this.inBalance.current.value))}/>}
        {(this.props.workmode=="Add")&&<input type="button" value="Add" id={"AddNew"} onClick={()=>voteEvents.emit("Eadd",parseInt(this.inId.current.value),this.inName.current.value,this.inFam.current.value,this.inOtch.current.value,parseInt(this.inBalance.current.value))}/>}
        <input type="button" value="Cancel" id={"Cancel"} onClick={()=>voteEvents.emit("EsetMode","Show",null)}/>
    </div>
}
        return ret;

    }

}

export default AddEditClient;
