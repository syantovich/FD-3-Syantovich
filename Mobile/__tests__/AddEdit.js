import React from 'react';
import renderer from 'react-test-renderer';

import MobileClient from "../components/MobileClient";
import AddEditClient from "../components/AddEditClient";

test('работа AddEdit', () => {

    const client = { id: 101, FIO: { fam: "Иванов", im: "Иван", otch: "Иванович" }, balance: 200 };

    const component = renderer.create(
        <AddEditClient
            client={client}
            workmode={"Edit"}
            edited={null}
            maxId={102}
        />
    );



    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();


    const buttonSave = component.root.findByProps({ id: "SaveEdit" })
    const buttonCancel = component.root.findByProps({ id: "Cancel" })

    buttonSave.props.onClick();
    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const buttonAdd=component.root.findByProps({ id: "AddItem" });
    buttonAdd.props.onClick();
    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();


    const buttonCreate = component.root.findByProps({ id: "AddNew" })

    buttonCreate.props.onClick();
    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

});