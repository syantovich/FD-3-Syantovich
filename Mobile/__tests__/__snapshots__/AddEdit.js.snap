// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа AddEdit 1`] = `
<div>
  id:
  <span>
    102
  </span>
  <br />
  <label>
    Имя:
    <input
      defaultValue={undefined}
      id="im"
      type="text"
    />
  </label>
  <br />
  <label>
    Фамилия:
    <input
      defaultValue={undefined}
      id="fam"
      type="text"
    />
  </label>
  <br />
  <label>
    Отчество:
    <input
      defaultValue={undefined}
      id="otch"
      type="text"
    />
  </label>
  <br />
  <label>
    Баланс:
    <input
      defaultValue={200}
      id="balance"
      type="number"
    />
  </label>
  <br />
  <input
    id="SaveEdit"
    onClick={[Function]}
    type="button"
    value="Save"
  />
  <input
    id="Cancel"
    onClick={[Function]}
    type="button"
    value="Cancel"
  />
</div>
`;
